storage/innobase/srv/srv0start.cc

```diff
@@ -2848,6 +2848,9 @@ innobase_start_or_create_for_mysql(void)
 	os_fast_mutex_free(&srv_os_test_mutex);
 
 	if (srv_rebuild_indexes) {
+		while (trx_rollback_or_clean_is_active) {
```

Comment:
I we were need volatile for extern variables used in loops like this one, InnoDB would probably stop working all together. Compiler will not optimize this loop because the call to `os_thread_sleep` can actually change any global variable as side effect. And there is a syscall inside of it, which makes things even more uncertain for compiler.